File: java/numbers-converter/src/Converter.java
public class Converter {

    protected int convertNumber(int number, int base) {
        int output = 0;
        int power = 0;

        while (number > 0) {
            int lastDigit = number % 10;
            output += lastDigit * Math.pow(base, power);
            number /= 10;
            power++;
        }
        return output;
    }

    protected int convertNumber(String number, int base) {
        int output = 0;
        int power = 0;
        char[] numberArray = number.toCharArray();

        for (int i = numberArray.length - 1; i >= 0; i--) {
            int currentNumber = numberArray[i];
            if (currentNumber - 48 <= 9) {
                output += (currentNumber - 48) * Math.pow(base, power);
            } else {
                output += (currentNumber - 55) * Math.pow(base, power);
            }
            power++;
        }
        return output;
    }
}

File: python/numbers-converter/converter.py
import math


def convert(number: int, base: int) -> int:
    output: int = 0
    power: int = 0

    while number > 0:
        current_digit: int = number % 10
        output += current_digit * math.pow(base, power)
        power += 1
        number = int(number / 10)

    return output


def convertHex(number: str, base: int) -> int:
    output: int = 0
    power: int = 0

    for digit in reversed(number):
        current_number = ord(digit) - 48 if digit.isdigit() else ord(digit) - 55
        output += current_number * math.pow(base, power)
        power += 1

    return output
